lib/services/cups.pm

```diff
@@ -18,31 +18,47 @@ use utils;
 use strict;
 use warnings;
 
+my $service_type = 'Systemd';
+
 sub install_service {
     zypper_call("in cups", exitcode => [0, 102, 103]);
 }
 
 sub config_service {
-    script_run 'echo FileDevice Yes >> /etc/cups/cups-files.conf';
-    validate_script_output 'cupsd -t', sub { m/is OK/ };
+    if ($service_type eq 'Systemd') {
+        script_run 'echo FileDevice Yes >> /etc/cups/cups-files.conf';
+        validate_script_output 'cupsd -t', sub { m/is OK/ };
+    } else {
+        # Allow new printers to be added using device URIs "file:/filename"
+        script_run 'echo FileDevice Yes >> /etc/cups/cupsd.conf';
+        assert_script_run 'cupsd';
```

Comment:
you do not need validate anything?

Reply:
I've run the test to run both paths. will run more tests to verify it. wait a moment. I'll fix the CI first.

Reply:
all cases passed and updated.